src/common/lib/pio_helper.h

```
@@ -22,9 +22,107 @@
+ * `@return` Allocated engine or {NULL, -1, -1} on failure.
+ */
+pio_engine_t claim_pio_and_sm(const pio_program_t* program);
+
+/**
+ * `@brief` Restart a PIO state machine at the given offset.
+ *
+ * `@param` pio    PIO instance.
+ * `@param` sm     State machine index.
+ * `@param` offset Program offset to jump to.
+ */
 void pio_restart(PIO pio, uint sm, uint offset);
```

Clang (14.0.6)

[warning] parameter name 'sm' is too short, expected at least 3 characters

(readability-identifier-length)